integration_tests/src/main/python/parquet_write_test.py

```diff
@@ -567,3 +568,76 @@ def test_write_empty_data_single_writer(spark_tmp_path):
     with_gpu_session(lambda spark: spark.createDataFrame(spark.sparkContext.parallelize(data), schema)
                      .write.mode("overwrite").partitionBy('c1', 'c2').parquet(data_path))
     with_cpu_session(lambda spark: spark.read.parquet(data_path).collect())
+
+
+PartitionWriteMode = Enum('PartitionWriteMode', ['Static', 'Dynamic'])
+
+
+def populate_partitioned_parquet_table(spark_tmp_table_factory, mode=PartitionWriteMode.Static):
+    """
+    Returns a function that does the following:
+        1. Creates an un-partitioned Hive table with data with the following schema:
+             1. make STRING
+             2. model STRING
+             3. year INT
+             4. type STRING (cardinality == 3)
+             5. comment STRING
+        2. Creates a partitioned Hive, with a similar schema to above, but partitioned on
+           the `type` column.
+        3. Populates the partitioned table using the un-partitioned table as input.
+             1. If the partition mode is dynamic, all the partitions are populated at once.
+             2. If the partition mode is static, the partitions are populated explicitly,
+                and sequentially.
+        4. Returns all the rows from 2 of the 3 partitions in the new table.
+    """
+
+    def create_input_table(spark):
+        tmp_input = spark_tmp_table_factory.get()
+        spark.sql("CREATE TABLE " + tmp_input +
+                  " (make STRING, model STRING, year INT, type STRING, comment STRING)" +
+                  " STORED AS PARQUET")
+        spark.sql("INSERT INTO TABLE " + tmp_input + " VALUES " +
+                  "('Ford',   'F-150',       2020, 'ICE',      'Popular' ),"
+                  "('GMC',    'Sierra 1500', 1997, 'ICE',      'Older'),"
+                  "('Chevy',  'D-Max',       2015, 'ICE',      'Isuzu?' ),"
+                  "('Tesla',  'CyberTruck',  2025, 'Electric', 'Fictional'),"
+                  "('Rivian', 'R1T',         2022, 'Electric', 'Heavy'),"
+                  "('Jeep',   'Gladiator',   2024, 'Hybrid',   'Upcoming')")
+        return tmp_input
+
+    def populate_partitions_static(spark):
+        input_table = create_input_table(spark)
+        output_table = spark_tmp_table_factory.get()
+        spark.sql("CREATE TABLE " + output_table +
+                  " (make STRING, model STRING, year INT, comment STRING)"
+                  " PARTITIONED BY (type STRING) STORED AS PARQUET")
+        spark.sql("INSERT INTO TABLE " + output_table + " PARTITION (type='ICE')" +
+                  " SELECT make, model, year, comment FROM " + input_table +
+                  " WHERE type='Electric'")
+        spark.sql("INSERT OVERWRITE TABLE " + output_table + " PARTITION (type='ICE')" +
+                  " SELECT make, model, year, comment FROM " + input_table +
+                  " WHERE type='Hybrid'")
```

Comment:
These are wrong, in multiple ways:
1. There should have been 3 `INSERT` statements here, one for each value of `type`.
2. The first insert for `ICE` should have been for `Electric`. The second should have been for `Hybrid`.
3. The final `SELECT` should not have capitalized `ELECTRIC`.

I'll fix this when I switch to using `assert_gpu_and_cpu_sql_writes_are_equal_collect()`.

Reply:
Fixed now.